packages/dataviews/src/dataviews-layouts/grid/index.tsx

```diff
@@ -285,24 +288,29 @@ function ViewGrid< Item >( {
 		{ regularFields: [], badgeFields: [] }
 	);
 	const hasData = !! data?.length;
-	const updatedPreviewSize = useUpdatedPreviewSizeOnViewportChange();
 	const hasBulkActions = useSomeItemHasAPossibleBulkAction( actions, data );
-	const usedPreviewSize = updatedPreviewSize || view.layout?.previewSize;
-	const gridStyle = usedPreviewSize
-		? {
-				gridTemplateColumns: `repeat(${ usedPreviewSize }, minmax(0, 1fr))`,
-		  }
-		: {};
+	const usedPreviewSize = view.layout?.previewSize;
+
+	// Calculate possible media sizes in grid for responsive images.
+	let sizes = '400px';
+	if ( usedPreviewSize ) {
+		// The max size the image can be is slightly less than twice its min size.
+		// Any bigger than that and the grid will start to show two images side by side.
+		// Sizes only needs a rough number so we don't count paddings and grid gap.
+		sizes = `${ usedPreviewSize * 2 }px`;
+	}
 	return (
 		<>
 			{ hasData && (
-				<Grid
-					gap={ 8 }
-					columns={ 2 }
-					alignment="top"
+				<div
 					className={ clsx( 'dataviews-view-grid', className ) }
-					style={ gridStyle }
+					style={ {
+						gridTemplateColumns:
+							usedPreviewSize &&
+							`repeat(auto-fill, minmax(${ usedPreviewSize }px, 1fr))`,
+					} }
 					aria-busy={ isLoading }
+					ref={ resizeObserverRef }
```

Comment:
As mentioned above, we need this ref here so DataViews can set the containerWidth in its context to the actual grid width, instead of the width of its wrapper that may or may not include scrollbars.

Reply:
(Also I changed this to use a `div` instead of `Grid` so I could pass it the ref)